### ISSUE TITLE
Community Metadata

### ISSUE THREAD

**Issue by elliotBraem** [near/devgigsboard#75]
This is related to [#491 Editing community leads to a 400 error message](https://github.com/near/neardevhub-widgets/issues/491) as a possible solution.

Assuming a completed #74, below is the **effective** Community Struct:

```rust
pub struct Community {
    pub admins: Vec<AccountId>,
    pub handle: CommunityHandle,
    pub name: String,
    pub tag: String,
    pub description: String,
    pub logo_url: String,
    pub banner_url: String,
    pub bio_markdown: Option<String>,
    pub github_handle: Option<String>,
    pub telegram_handle: Vec<String>,
    pub twitter_handle: Option<String>,
    pub website_url: Option<String>,
    pub addons: Vec<CommunityAddOn>,
}
```

Currently, there is an issue when we are updating communities that the payload is too large. This is because our only available [update_community](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L512) function will update the entire community at a time (including all of the addons and their data). This payload is too large for the wallet redirect and so transactions are failing.

A way to fix this is splitting the available update functions, I'm thinking:

- [ ] update_community_metadata
- [ ] update_community_admins
- [x] set_community_addons (exists)

And in this migration to metadata, I would propose that we reflect the [social contract metadata standard](https://github.com/NearSocial/standards/blob/main/types/common/Metadata.md) more:

```rust
pub struct Community {
    pub handle: CommunityHandle, <-- currently this is editable, I do not think it should be
    pub admins: Vec<AccountId>,
    pub metadata: CommunityMetadata,
    pub addons: Vec<CommunityAddOn>,
}

pub struct CommunityMetadata {
    pub name: String,
    pub description: String,
    pub image: String, 
    pub backgroundImage: String,
    pub linktree: Vec<String>    <--- Key value pairs, could be stringified objects or looking for suggestions
    pub tags: Vec<String>,
    pub bio: String
}
```

Splitting into these functions should solve the issue, as well as more closely reflecting social db metadata standards will enable template swapping and more widget support, such as mob.near/widget/Image, the metadata editor, and more.

---
Note:

It seems that there had been an attempt to use CommunityMetadata before; although these remnant's exist in the code, they are not used besides in [get_community_metadata](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L373C12-L373C34) and [get_all_communities_metadata](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L403), although it's only an object to be mapped to.

Accepting this change will require several changes on the front end to use the revised structure and functions.

**Comment by Tguntenaar** [comment]
Hey @elliotBraem ! 

It seems that the images, the tags and the linktree property are all objects with key value pairs. Is there a reason you chose to go for the String type? 

We can reflect these properties as Maps. 

```
{
  "name": "Near Social",
  "image": {
    "ipfs_cid": "bafkreid55mbassw335wezjk3gc5rzizwrqpenfxk3dyyljumxvezplhjg4"
  },
  "linktree": {
    "twitter": "NearSocial_",
    "github": "NearSocial",
    "telegram": "NearSocial",
    "website": "near.social"
  },
  "description": "# Near Social\n\n**Near Social** is ...",
  "backgroundImage": {
    "ipfs_cid": "bafybeigwqqx2eccpyip66bn2bn6janvss2snv3xuuus3ltin6i46mfkyam"
  },
  "tags": {
    "platform": "",
    "project": "",
    "network": "",
    "protocol": "",
    "open-source": ""
  }
}
```

**Comment by Tguntenaar** [comment]
@ori-near I noticed we are making a difference between `bio_markdown` and `description` right now, which I'm not sure why. In the [social contract metadata standard](https://github.com/NearSocial/standards/blob/main/types/common/Metadata.md) `description` is also a string in markdown format. 

I've looked at a few communities and the once I saw are repeating more or less the same message in these fields. By the way, `bio_markdown` is visible in the sidebar on the activity tab. 

For now, I will store it in the Community struct instead of the metadata @elliotBraem.

**Comment by Tguntenaar** [comment]
Hey @elliotBraem, I just discussed with @frol that restructuring the community metadata will be double work since this will also be fixed by [#540](https://github.com/near/neardevhub-widgets/issues/540). 

Also splitting up the update_community funciton is not effective for issue [#491](https://github.com/near/neardevhub-widgets/issues/491). This is because set_community_addons is already been split up and will give the error as well. We will have to wait untill the old wallet is deprecated.

Updating the addons name can be done from the addons configurator directly. But currently, the order of the addons won't be able to change in chases where the vector of addons is already too big in size.

So this issue can be closed.

**Comment by frol** [comment]
Just to clarify, after https://github.com/near/neardevhub-widgets/issues/540 we should just set the metadata to SocialDB.
